projects/gateway/pkg/syncer/translator_syncer_integration_test.go

```diff
@@ -0,0 +1,222 @@
+				Routes: []*v1.Route{
+					{
+						Matchers: []*matchers.Matcher{
+							{
+								PathSpecifier: &matchers.Matcher_Prefix{Prefix: "/"},
+							},
+						},
+						Action: &v1.Route_DirectResponseAction{
+							DirectResponseAction: &gloov1.DirectResponseAction{
+								Status: 200,
+								Body:   "foo",
+							},
+						},
+					},
+				},
+			},
+		}
+		_, err = baseVirtualServiceClient.Write(vs, clients.WriteOpts{})
+		Expect(err).NotTo(HaveOccurred())
+		gw := &v1.Gateway{
+			GatewayType: &v1.Gateway_HttpGateway{
+				HttpGateway: &v1.HttpGateway{
+					VirtualServices: []core.ResourceRef{
+						vs.Metadata.Ref(),
+					},
+				},
+			},
+			BindAddress: "::",
+			BindPort:    8080,
+			ProxyNames:  []string{"gateway-proxy"},
+			Metadata: core.Metadata{
+				Name:      "gateway-proxy",
+				Namespace: "gloo-system",
+			},
+		}
+		_, err = gatewayClient.Write(gw, clients.WriteOpts{})
+		Expect(err).NotTo(HaveOccurred())
+		snapshot = func() *v1.ApiSnapshot {
+			vss, err := baseVirtualServiceClient.List("gloo-system", clients.ListOpts{})
+			Expect(err).NotTo(HaveOccurred())
+			gws, err := gatewayClient.List("gloo-system", clients.ListOpts{})
+			Expect(err).NotTo(HaveOccurred())
+			return &v1.ApiSnapshot{
+				VirtualServices: vss,
+				Gateways:        gws,
+			}
+		}
+
+	})
+
+	AfterEach(func() {
+		cancel()
+	})
+
+	EventuallyProxyStatusInVs := func() gomega.AsyncAssertion {
+		return Eventually(func() (core.Status_State, error) {
+			newvs, err := baseVirtualServiceClient.Read(vs.Metadata.Namespace, vs.Metadata.Name, clients.ReadOpts{})
+			if err != nil {
+				return core.Status_Pending, err
+			}
+			subresouce := newvs.GetStatus().SubresourceStatuses
+			if subresouce == nil {
+				return core.Status_Pending, fmt.Errorf("no status")
+			}
+			proxyState := subresouce["*v1.Proxy.gloo-system.gateway-proxy"]
+			if proxyState == nil {
+				return core.Status_Pending, fmt.Errorf("no state")
+			}
+			return proxyState.State, nil
+		})
+	}
+	EventuallyProxyStatus := func() gomega.AsyncAssertion {
+		return Eventually(func() (core.Status_State, error) {
+			proxy, err := proxyClient.Read("gloo-system", "gateway-proxy", clients.ReadOpts{})
+			if err != nil {
+				return core.Status_Pending, err
+			}
+			return proxy.Status.State, nil
+		})
+	}
+
+	AcceptProxy := func() {
+		proxy, err := proxyClient.Read("gloo-system", "gateway-proxy", clients.ReadOpts{})
+		Expect(err).NotTo(HaveOccurred())
+		proxy.Status = core.Status{State: core.Status_Accepted}
+		proxyClient.Write(proxy, clients.WriteOpts{OverwriteExisting: true})
+	}
+
+	It("should set status correctly even when the status from the snapshot was not updated", func() {
+
+		ts.Sync(ctx, snapshot())
+		// wait for proxy to be written
+		Eventually(func() (bool, error) {
+			_, err := proxyClient.Read("gloo-system", "gateway-proxy", clients.ReadOpts{})
+			if err != nil {
+				return false, err
+			}
+			return true, nil
+		}).Should(BeTrue())
```

Comment:
we can simplify this to:
```suggestion
		Eventually(func() (bool, error) {
			return proxyClient.Read("gloo-system", "gateway-proxy", clients.ReadOpts{})
		}).ShouldNot(BeNil())
```